datasetUtils/checkForNewVersion_test.go

```diff
@@ -0,0 +1,102 @@
+package datasetUtils
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"bytes"
+	"log"
+)
+
+func TestFetchLatestVersion(t *testing.T) {
+	// Create a mock HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Return a JSON response similar to the GitHub API
+		w.Write([]byte(`{"tag_name": "v1.0.0"}`))
+	}))
+	defer server.Close()
+	
+	// Replace GitHubAPI with the URL of the mock server
+	oldGitHubAPI := GitHubAPI
+	GitHubAPI = server.URL
+	defer func() { GitHubAPI = oldGitHubAPI }()
+	
+	// Create a mock HTTP client
+	client := server.Client()
+	
+	// Call fetchLatestVersion
+	version, err := fetchLatestVersion(client)
+	if err != nil {
+		t.Fatalf("Expected no error, got %v", err)
+	}
+	
+	// Check the version number
+	if version != "v1.0.0" {
+		t.Errorf("Expected version v1.0.0, got %s", version)
+	}
+}
+
+func TestCheckForNewVersion(t *testing.T) {
+	tests := []struct {
+		name           string
+		currentVersion string
+		mockResponse   string
+		expectedLog    string
+		}{
+			{
+				name:           "New version available",
+				currentVersion: "0.9.0",
+				mockResponse:   `{"tag_name": "1.0.0"}`,
```

Comment:
should this be **v**1.0.0?

Reply:
Doesn't really matter, the code strips the v.

Reply:
yes, but the fact that it is stripped is part of the logic to test